front/app/src/Views/TestBall/TestBall.tsx

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable max-lines-per-function */
+/* eslint-disable max-statements */
+import { useEffect, useRef, useState } from "react";
+
+import Game from "./Objects/Game";
+
+import { io } from "socket.io-client";
+
+const	URL = "http://localhost:3000";
+
+const	TestBall = () =>
+{
+	const
+	[
+		connected,
+		setConnected
+	] = useState(false);
+	const game = new Game();
+	game.board.game = game;
+
+	game.ball.game = game;
+	game.net.game = game;
+
+	const	canvasRef = useRef<HTMLCanvasElement>(null);
+	game.board.canvasRef = canvasRef;
+	const	update = () =>
+	{
+		game.playerOne.updatePlayerPosition();
+		game.playerTwo.updatePlayerPosition();
+		game.ball.update();
+	};
+
+	const clear = () =>
+	{
+		if (game.board.ctx)
+		{
+			game.board.ctx.fillStyle = "#fff";
+			game.board.ctx?.clearRect(0, 0,
+				game.board.dim.width, game.board.dim.height);
+		}
+	};
+
+	const	keyHookDown = (e: KeyboardEvent) =>
+	{
+		switch (e.code)
+		{
+			case "ArrowUp":
+				game.actionKeyPress = 38;
+				break;
+			case "ArrowDown":
+				game.actionKeyPress = 40;
+				break;
+			case "KeyS":
+				game.actionKeyPress = 83;
+				break;
+			case "KeyW":
+				game.actionKeyPress = 87;
+				break;
+			default:
+				break;
+		}
+	};
+
+	const	keyHookReleased = () =>
+	{
+		game.actionKeyPress = -1;
+	};
+
+	const keyEnter = () =>
+	{
+		game.continueAnimating = true;
+		game.startDisplayed = false;
+	};
+
+	const pauseButtonRef = useRef<HTMLInputElement>(null);
+	const resumeButtonRef = useRef<HTMLInputElement>(null);
+
+	useEffect(() =>
+	{
+		const socket = io(URL,
+		{
+			autoConnect: false,
+			reconnectionAttempts: 5,
+		});
+
+		const connect = () =>
+		{
+			console.log("ws connected");
+			setConnected(true);
+		};
+
+		const disconnect = () =>
+		{
+			console.log("ws disconnected");
+			setConnected(false);
+		};
+
+		const	connectError = (error: Error) =>
+		{
+			console.error("ws_connect_error", error);
+		};
+
+		socket.on("connect", connect);
+		socket.on("disconnect", disconnect);
+		socket.on("error", connectError);
+		socket.connect();
+		return (() =>
+		{
+			socket.off("connect", connect);
+			socket.off("disconnect", disconnect);
+			socket.off("error", connectError);
+		});
+	}, []);
```

Comment:
Nota Bene : the empty array : [] is intentionnaly here, there means  no dependency to the effect.
The useEffect is setted like a lifecycle function 
In this configuration, the use effect is called only one time after the component is mounted.